src/acl/external/kerberos_ldap_group/support_krb5.cc

```diff
@@ -290,6 +305,11 @@ krb5_create_cache(char *domain)
                 } else {
                     debug((char *) "%s| %s: DEBUG: Found principal name: %s\n", LogTime(), PROGRAM, principal_name);
                     found = 1;
+                    if (service_principal_name != NULL && strcasecmp(principal_name,service_principal_name)) {
```

Comment:
no need for the "!= NULL" part here, and the str*() function should have "!= 0" for clarity